source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -90,8 +90,9 @@ Http::FilterHeadersStatus CacheFilter::encodeHeaders(Http::ResponseHeaderMap& he
 
   if (filter_state_ == FilterState::ValidatingCachedResponse && isResponseNotModified(headers)) {
     processSuccessfulValidation(headers);
-    // Stop the encoding stream until the cached response is fetched & added to the encoding stream.
-    return Http::FilterHeadersStatus::StopIteration;
+    // Continue encoding the headers but do not end the stream as the response body is yet to be
+    // injected.
+    return Http::FilterHeadersStatus::ContinueAndDontEndStream;
```

Comment:
Does this correctly handle the case where the cached response is a header only response?